cpp/include/raft/neighbors/ivf_pq_helpers.cuh

```diff
@@ -407,5 +575,78 @@ void erase_list(raft::resources const& res, index<IdxT>* index, uint32_t label)
   ivf_pq::detail::erase_list(res, index, label);
 }
 
+/**
+ * @brief Public helper API for computing the index's rotation matrix.
+ *
+ * @tparam IdxT
+ * @param[in] res
+ * @param[inout] index
+ * @param[in] force_random_rotation whether to apply a random rotation matrix on the input data. See
+ * raft::neighbors::ivf_pq::index_params for more details.
+ */
+template <typename IdxT>
+void make_rotation_matrix(raft::resources const& res,
+                          index<IdxT>* index,
+                          bool force_random_rotation)
+{
+  raft::neighbors::ivf_pq::detail::make_rotation_matrix(res,
+                                                        force_random_rotation,
+                                                        index->rot_dim(),
+                                                        index->dim(),
+                                                        index->rotation_matrix().data_handle());
+}
+
+/**
+ * @brief Public helper API for externally modifying the index's IVF centroids.
```

Comment:
Can you just briefly explain the implications of using this function? For example- does this re-cluster any data points that are already in the given index? Or does it assume the index is empty? Do we assert this? Please also provide a brief usage example (see above). Further, if not already, these functinos should be in the docs under their own group / heading for `IVF PQ Helpers`.

Reply:
Done.